core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala

```diff
@@ -77,6 +77,30 @@ class AppStatusStoreSuite extends SparkFunSuite {
     assert(store.count(classOf[CachedQuantile]) === 2)
   }
 
+  test("only successfull task have taskSummary") {
+    val store = new InMemoryStore()
+    (0 until 5).foreach { i => store.write(failedTaskData(i)) }
+    val appStore = new AppStatusStore(store).taskSummary(stageId, attemptId, uiQuantiles)
+    assert(appStore.size === 0)
+  }
+
+  test("summary should contain task metrics of only successfull tasks") {
+    val store = new InMemoryStore()
+    (0 until 5).foreach { i =>
+      if (i % 2 == 1) store.write(failedTaskData(i))
+      else store.write(newTaskData(i))
+    }
+    val summary = new AppStatusStore(store).taskSummary(stageId, attemptId, uiQuantiles).get
+
+    val values = (0 to 2).map( i =>
```

Comment:
2.0 is much clearer than 2.toDouble. But why not just write `Array(0.0, 2.0, 4.0)`?

Reply:
Done